docs/src/main/asciidoc/deploying-to-openshift.adoc

```diff
@@ -1,281 +1,420 @@
+
+NOTE: Use the Quarkus OpenShift extension for development and testing purposes only. For production environments, consider using the traditional S2I method described in xref:proc-using-s2i-openshift-quarkus_{context}[].
 
-The aforementioned command will build a jar file locally, trigger a container image build and then apply the generated OpenShift resources.
-The generated resources are using OpenShift's `DeploymentConfig` that is configured to automatically trigger a redeployment when a change in the `ImageStream` is noticed.
-In other words, any container image build after the initial deployment will automatically trigger redeployment, without the need to delete, update or re-apply the generated resources.
+.Procedure
 
-To confirm the above command has created an image stream, a service resource and has deployed the application (has a pod running), apply these commands:
+. Change to the directory that contains your Quarkus Maven project.
 
-[source,bash,subs=attributes+]
+. To add the OpenShift extension to an existing project, enter the following command:
++
+[source, subs=attributes+]
 ----
-oc get is
-oc get pods
-oc get svc
+./mvnw quarkus:add-extension -Dextensions="openshift"
 ----
-
-To expose the created service to a route and test it:
-
-[source,bash,subs=attributes+]
++
+NOTE: You can include the `-Dextensions="openshift"` argument to add the Quarkus OpenShift extension when you create a new project.
++
+When you add the OpenShift extension, the script adds the following dependency to the `pom.xml` file:
++
+[source,xml]
 ----
-oc expose svc/greeting
-oc get routes
-curl http://<route>/greeting
+   <dependency>
+     <groupId>io.quarkus</groupId>
+     <artifactId>quarkus-openshift</artifactId>
+   </dependency>
+----
+. If you are using an untrusted certificate, add the following line to the `src/main/resources/application.properties` file:
++
+[source]
+----
+quarkus.kubernetes-client.trust-certs=true
 ----
 
-=== Customizing
-
-All available customization options are available in the link:deploying-to-kubernetes#openshift[OpenShift configuration options].
-
-Some examples are provided in the sections below:
-
-==== Exposing Routes
-
-To expose a `Route` for the Quarkus application:
-
-[source,properties]
+. To direct OpenShift to use the Open JDK {JDKVersion} {RHEL} 7 image, add the following line to the `application.properties` file:
++
+[source,xml]
+----
+quarkus.s2i.base-jvm-image=registry.access.redhat.com/openjdk/openjdk-11-rhel7
+----
++
+NOTE: If you are deploying on IBM Z infrastructure, add `quarkus.s2i.base-jvm-image=registry.access.redhat.com/openj9/openj9-11-rhel8` to the `application.properties` file.
```

Comment:
Is it intended that this end up in community docs as well? I am fairly certain that there is very little community interest in IBM Z.